dll/win32/newdev/newdev.c

```diff
@@ -366,12 +372,20 @@ SearchDriverRecursive(
         ok = FindNextFileW(hFindFile, &wfd))
     {
 
+        /* Filename is too long to be searched */
+        if (wcslen(wfd.cFileName) >= _countof(FileName))
+            continue;
+
         wcscpy(FileName, wfd.cFileName);
         if (IsDots(FileName))
             continue;
 
         if (wfd.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
         {
+            /* Full path is too long to be searched */
+            if (wcslen(DirPath) + wcslen(FileName) >= _countof(FullPath))
+                continue;
+
             /* Recursive search */
             wcscpy(FullPath, DirPath);
             wcscat(FullPath, FileName);
```

Comment:
```suggestion
            /* Recursive search */
            hr = StringCchPrintfW(FullPath, _countof(FullPath), L"%s%s", DirPath, FileName);
            if (!SUCCEEEDED(hr))
            {
                /* Full path is too long to be searched */
                continue;
            }
```
(And see the comment from above, where `FileName` could be replaced by `wfd.cFileName`.)

Reply:
I have replaced FileName by wfd.cFileName and removed FileName local variable.